engine/src/dot/http_rpc.rs

```diff
@@ -70,6 +72,46 @@ impl RpcClientT for PolkadotHttpClient {
 	}
 }
 
+/// Adds a default port to the url based on the scheme (http, https, ws, wss),
+/// if none exists. Otherwise preservers existing port.
+///
+/// This function assumes that `url` is already validated, i.e.:
+///  - It's accepted by `Url::parse()`
+///  - It includes a host part
+///  - It does not have a fragment part
+fn ensure_port(url: SecretUrl) -> Result<SecretUrl> {
+	// we use url::Url to get the default port for our scheme
+	let targetport = Url::parse(url.as_ref())
+		.unwrap() // We know that `url` is valid
+		.port_or_known_default()
+		.ok_or(anyhow::anyhow!("Unknown scheme and no port given."))?;
+
+	// We use http::uri::Uri to ensure that the default port is added if none exists
+	// We split the uri into components, insert the port, and reconstruct.
+	let mut parts = url
+		.as_ref()
+		.parse::<Uri>()
+		.unwrap() // We know that `url` is valid
+		.into_parts();
+
+	// Update the authority part of the uri by mapping over it.
+	let authority = parts.authority.unwrap();
```

Comment:
Should throw error instead of panicking here, or expect() with a reason why it's safe
```suggestion
	let authority = parts.authority.unwrap();
```

Reply:
Yes, `expect` makes sense. Updated all occurences of unwrap.